network-config-analyzer/IstioPolicyYamlParser.py

```diff
@@ -328,12 +329,12 @@ def parse_operation(self, operation_dict):
         self.validate_dict_elem_has_non_empty_array_value(operation, 'to.operation')
 
         dst_ports = self.get_rule_ports(operation.get('ports'), operation.get('notPorts'))  # PortSet
-        methods_dfa = self.parse_regex_dimension_values("methods", operation.get("methods"),
+        methods_dfa = self.parse_regex_dimension_values("methods_dfa", operation.get("methods"),
```

Comment:
we should not call `parse_regex_dimension_values` for the methods dimension. The goal is to avoid computation of MinDFA for input method values, and instead check each input value, whether it matches any of the allowed method values (if it's a regex, use python regex matching, and if it's a simple string value just use str comparison) , and hold/create the relevant MethodSet accordingly. An initial version for this implementation is already in the old branch referred in the issue itself: 
https://github.com/IBM/network-config-analyzer/blob/46ec24e789e7c2089f28c52a5a3c5adee49787e3/network-config-analyzer/IstioPolicyYamlParser.py#L287-L301 
https://github.com/IBM/network-config-analyzer/blob/46ec24e789e7c2089f28c52a5a3c5adee49787e3/network-config-analyzer/IstioPolicyYamlParser.py#L229-L238